drivers/sensor/sunxi-thermal.c

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright © 2017-2018 The Crust Firmware Authors.
+ * SPDX-License-Identifier: (BSD-3-Clause OR GPL-2.0)
+ */
+
+#include <debug.h>
+#include <error.h>
+#include <limits.h>
+#include <mmio.h>
+#include <sensor.h>
+#include <sensor/sunxi-thermal.h>
+
+#define SENSOR_COUNT        3
+
+#define THS_0_DATA_REGISTER 0x80
+#define THS_1_DATA_REGISTER 0x84
+#define THS_2_DATA_REGISTER 0x88
+
+enum {
+	THS0,
+	THS1,
+	THS2,
+};
+
+struct sensor_info sunxi_thermal_sensor_info[SENSOR_COUNT] = {
+	[THS0] = {
+		.name    = "ths0",
+		.reading = UINT32_MAX, /* Symbolizes no reading yet. */
+		.address = 0x80,
+	},
+	[THS1] = {
+		.name    = "ths1",
+		.reading = UINT32_MAX, /* Symbolizes no reading yet. */
+		.address = 0x84,
+	},
+	[THS2] = {
+		.name    = "ths2",
+		.reading = UINT32_MAX, /* Symbolizes no reading yet. */
+		.address = 0x88,
+	},
+};
+
+static uint8_t
+sunxi_thermal_get_count(struct device *dev __unused)
+{
+	return SENSOR_COUNT;
+}
+
+static struct sensor_info *
+sunxi_thermal(struct device *dev __unused, uint8_t id)
+{
+	assert(id < SENSOR_COUNT);
+
+	return &sunxi_thermal_sensor_info[id];
+}
+
+static int
+sunxi_thermal_read_raw(struct device *dev, uint8_t id, uint32_t *raw __unused)
+{
+	uint32_t value;
+
+	value =
+		mmio_read32(dev->regs +
+		            sunxi_thermal_sensor_info[id].address) &
+		BITMASK(0, 11);
+
+	sunxi_thermal_sensor_info[id].reading = value;
+
+	return SUCCESS;
+}
+
+static int
+sunxi_thermal_probe(struct device *dev)
+{
+	if (!dev)
```

Comment:
No need for this, as a valid device is always given to the probe function.

Reply:
without it, it throws a warning for dev not being used

Reply:
`__unused` (look at `get_info`)

Reply:
if it do that then that will cause an error on the assert statement

Reply:
I stand corrected. The change has been made

Reply:
No it won't. See https://github.com/crust-firmware/crust/blob/master/drivers/clock/sunxi-ccu.c#L262